app/APIStart.py

```diff
@@ -8,148 +8,206 @@
+
+
+@app.route('/item/search/<string:loc>', methods=['GET'])
 def search_item(loc):
     try:
-        testdb = db()
-        resultlist = testdb.search_item_by_loc(loc)
-        json_string = json.dumps([ob.__dict__ for ob in resultlist])
-        return json_string
+        items = db.session.query(Item).filter(Item.location == loc).all()
+        db.session.close()
+        return jsonify(data=[item.serialize() for item in items])
     except Exception as e:
-        print(e)
+        resp = jsonify({"error": e.__str__()})
+        resp.status_code = 200
+        return resp
 
-@app.route('/item/search/<string:name>',methods=['PUT'])
+
+@app.route('/item/search/<string:name>', methods=['PUT'])
 def search_item_name(name):
     try:
-        testdb = db()
-        resultlist = testdb.search_item_by_name(name)
-        json_string = json.dumps([ob.__dict__ for ob in resultlist])
-        return json_string
+        items = db.session.query(Item).filter(Item.item_name == name).all()
+        db.session.close()
+        return jsonify(data=[item.serialize() for item in items])
     except Exception as e:
-        print(e)
-    
-@app.route('/user/register',methods=['POST'])
+        resp = jsonify({"error": e.__str__()})
+        resp.status_code = 200
+        return resp
+
+
+@app.route('/user/register', methods=['POST'])
 def register_user():
     try:
-        testdb = db()
-        username =request.json['username'] 
+        username = request.json['username']
         email = request.json['email']
-        password = request.json['password']
-        user = User()
-        user.User(username,email,password)
-        
-        excep=testdb.register_user(user)
-        
-        if excep:
-            resp = jsonify({"Action":'Error occured {}'.format(excep)})
-            resp.status_code = 200
-            return resp
-        else:
-            resp = jsonify({"Action":'User Registered Successfully'})
-            resp.status_code = 200
-            return resp        
-        
-        
+        passw = request.json['password']
+
+        new_user = User(username, email, passw)
+
+        db.session.add(new_user)
+        db.session.commit()
+        db.session.close()
+
+        resp = jsonify({"Action": 'User Registered Successfully'})
+        resp.status_code = 200
+        return resp
+
     except Exception as e:
-        print(e)
+        resp = jsonify({"error": e.__str__()})
+        resp.status_code = 200
+        return resp
 
 
-@app.route('/user/login',methods=['POST'])
+@app.route('/user/login', methods=['POST'])
 def login_user():
     try:
-        testdb = db() 
         email = request.json['email']
-        password = request.json['password']
-        
-        if testdb.login_user(email,password):
-            resp = jsonify({"Action":'User Logged In Successfully'})
+        passw = request.json['password']
+        check = db.session.query(User).filter(User.email == email).filter(User.password == passw).all()
```

Comment:
this query looks bit off, you can have multiple fields in **filter**

Reply:
Ok will Change it. Based on my query i found this syntax earlier